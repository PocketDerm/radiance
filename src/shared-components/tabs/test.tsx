import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { Tabs } from './index';

describe('<Tabs />', () => {
  describe('UI snapshot', () => {
    it('renders with default props', () => {
      const { container } = render(
        <Tabs
          tabItems={[
            { id: 1, text: 'Tab 1' },
            { id: 2, text: 'Tab 2' },
            { id: 3, text: 'Tab 3' },
          ]}
        />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });

    it('renders with initialActiveTabId and onClick props', () => {
      const { container } = render(
        <Tabs
          initialActiveTabId={2}
          onClick={() => undefined}
          tabItems={[
            { id: 1, text: 'Tab 1' },
            { id: 2, text: 'Tab 2' },
            { id: 3, text: 'Tab 3' },
          ]}
        />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('on click function', () => {
    test('returns correct params', () => {
      const spy = jest.fn();
      const { container } = render(
        <Tabs
          initialActiveTabId={1}
          tabItems={[
            { id: 1, text: 'Tab 1' },
            { id: 2, text: 'Tab 2' },
            { id: 3, text: 'Tab 3' },
          ]}
          onClick={spy}
        />,
      );

      container.querySelector('button')?.click();

      expect(spy).toHaveBeenCalledWith({ id: 1, text: 'Tab 1' });
    });
  });
});
