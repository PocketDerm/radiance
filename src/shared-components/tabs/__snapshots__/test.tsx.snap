// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> UI snapshot renders with default props 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin: 2.5rem 0 0 0;
  position: relative;
  width: 100%;
}

@media (min-width:768px) {
  .emotion-6 {
    margin: 3.5rem 0 0 0;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-0 {
  background: transparent;
  border: none;
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin: 0;
  padding: 1rem 1.5rem;
  -webkit-transition: 350ms;
  transition: 350ms;
  border-radius: 4px;
  color: #332E54;
}

.emotion-0:hover {
  color: #332E54;
  -webkit-transition: 350ms;
  transition: 350ms;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-2 {
  background: transparent;
  border: none;
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin: 0;
  padding: 1rem 1.5rem;
  -webkit-transition: 350ms;
  transition: 350ms;
  border-radius: 4px;
  color: #C3C0CD;
}

.emotion-2:hover {
  color: #332E54;
  -webkit-transition: 350ms;
  transition: 350ms;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

<div
  class="emotion-6 emotion-7"
>
  <button
    class="emotion-0 emotion-1"
  >
    Tab 1
  </button>
  <button
    class="emotion-2 emotion-1"
  >
    Tab 2
  </button>
  <button
    class="emotion-2 emotion-1"
  >
    Tab 3
  </button>
</div>
`;

exports[`<Tabs /> UI snapshot renders with initialActiveTabId and onClick props 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin: 2.5rem 0 0 0;
  position: relative;
  width: 100%;
}

@media (min-width:768px) {
  .emotion-6 {
    margin: 3.5rem 0 0 0;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-2 {
  background: transparent;
  border: none;
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin: 0;
  padding: 1rem 1.5rem;
  -webkit-transition: 350ms;
  transition: 350ms;
  border-radius: 4px;
  color: #332E54;
}

.emotion-2:hover {
  color: #332E54;
  -webkit-transition: 350ms;
  transition: 350ms;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-0 {
  background: transparent;
  border: none;
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
  font-weight: bold;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin: 0;
  padding: 1rem 1.5rem;
  -webkit-transition: 350ms;
  transition: 350ms;
  border-radius: 4px;
  color: #C3C0CD;
}

.emotion-0:hover {
  color: #332E54;
  -webkit-transition: 350ms;
  transition: 350ms;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

<div
  class="emotion-6 emotion-7"
>
  <button
    class="emotion-0 emotion-1"
  >
    Tab 1
  </button>
  <button
    class="emotion-2 emotion-1"
  >
    Tab 2
  </button>
  <button
    class="emotion-0 emotion-1"
  >
    Tab 3
  </button>
</div>
`;
